### ISSUE TITLE
Change Code Analyzer For Constructor's Private Member Generation

### ISSUE THREAD

**Issue by kspearrin** [dotnet/roslyn#21676]
**Version Used**: VS 2017 15.3

**Steps to Reproduce**:

VS 2017 15.3 gives me this handy little constructor shortcut to generate a `private readonly` field and assign it. I think these are the code analyzers that come from Roslyn??

Screenshot:

[![enter image description here][1]][1]

This ends up generating a private member called `userService` and then assigns it with:

    this.userService = userService;

This goes against the code style that I use which is to name all private members with a prefix `_` resulting in assignment that should look like:

    _userService = userService;

Is it possible to change this analyzer so that VS obeys my code style?

  [1]: https://i.stack.imgur.com/L3Ec9.png

**Expected Behavior**:

Generate private member with `_` prefix.

**Actual Behavior**:

Private member is generated with no prefix and `this.` is used for assignment in the constructor.

**Comment by Pilchie** [comment]
The name chosen should follow the naming conventions that you have configured (either in a `.editorconfig` file, or Tools\Options\Text Editor\C#\Code Style\Naming), for example.  Right @CyrusNajmabadi

**Comment by kspearrin** [comment]
What is the .editorconfig flag for a underscore prefix on private members? I don't think I have that one set.

**Comment by Pilchie** [comment]
See https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference for the general docs on specifying naming styles in .editorconfig.

**Comment by kspearrin** [comment]
@Pilchie Thanks. I've referenced this documentation and come up with the following:

```
[*.{cs,vb}]
dotnet_naming_rule.async_methods_end_in_async.symbols  = private_members
dotnet_naming_rule.async_methods_end_in_async.style    = underscore_prefix
dotnet_naming_rule.async_methods_end_in_async.severity = suggestion

dotnet_naming_symbols.private_members.applicable_kinds           = field
dotnet_naming_symbols.private_members.applicable_accessibilities = private
dotnet_naming_symbols.private_members.required_modifiers         = readonly

dotnet_naming_style.underscore_prefix.required_prefix = _
```

But this seems to be crashing my visual studio now whenever I open a `.cs` file to edit. Happened 3 times until I removed it.

Any ideas? The docs are a little hard to follow so I may be doing something wrong here.

**Comment by Pilchie** [comment]
There is a known bug (#19685) when something is partially specified.  @dpoeschl would be the authoritative source here, but is on vacation this week.  In the meantime, @kuhlenh or @jmarolf might be able to help.

Maybe you need to include the other fields of the `dotnet_naming_style.underscore_prefix.*` ?

Note that the `async_methods_end_in_async` part there is the "name" of the style, and might be better renamed to `private_fields_start_with_underscore`.

**Comment by kspearrin** [comment]
Yep, I noticed the name after I commented. Updated.

**Comment by kspearrin** [comment]
@Pilchie This one worked for me.

```
dotnet_naming_rule.private_members_with_underscore.symbols  = private_members
dotnet_naming_rule.private_members_with_underscore.style    = underscore_prefix
dotnet_naming_rule.private_members_with_underscore.severity = suggestion

dotnet_naming_symbols.private_members.applicable_kinds           = field
dotnet_naming_symbols.private_members.applicable_accessibilities = private
dotnet_naming_symbols.private_members.required_modifiers         = readonly

dotnet_naming_style.underscore_prefix.capitalization = camel_case
dotnet_naming_style.underscore_prefix.required_prefix = _
dotnet_naming_style.underscore_prefix.required_suffix = 
dotnet_naming_style.underscore_prefix.word_separator = 
```

**Comment by CyrusNajmabadi** [comment]
Is there still a bug then?  Thanks!

**Comment by kspearrin** [comment]
I don't think so. Closing.
